server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -57,8 +57,21 @@
         .toFormatter(Locale.ROOT)
         .withResolverStyle(ResolverStyle.STRICT);
 
+    private static final DateTimeFormatter STRICT_YEAR_MONTH_DAY_PRINTER = new DateTimeFormatterBuilder()
+        .appendValue(ChronoField.YEAR, 4, 9, SignStyle.EXCEEDS_PAD)
+        .optionalStart()
+        .appendLiteral("-")
+        .appendValue(MONTH_OF_YEAR, 2, 2, SignStyle.NOT_NEGATIVE)
+        .optionalStart()
+        .appendLiteral('-')
+        .appendValue(DAY_OF_MONTH, 2, 2, SignStyle.NOT_NEGATIVE)
+        .optionalEnd()
+        .optionalEnd()
+        .toFormatter(Locale.ROOT)
+        .withResolverStyle(ResolverStyle.STRICT);
+
     private static final DateTimeFormatter STRICT_YEAR_MONTH_DAY_FORMATTER = new DateTimeFormatterBuilder()
-        .appendValue(ChronoField.YEAR, 4, 10, SignStyle.EXCEEDS_PAD)
+        .appendValue(ChronoField.YEAR, 4, 4, SignStyle.EXCEEDS_PAD)
```

Comment:
Isn't this a breaking change? And it then limits the ability to use dates in the future (yes past 9999)? Is the expectation then that dates in the distant future must use epoch seconds or millis?

Reply:
we have accidentally changed  the max length of year in strict_date_optional_time in 7.0.
If we do not consider this accidental change a bug, then agree it would be a breaking change. 
This change - limiting the length from 10 to 4 - might help parsing epoch millis with length between 4 and 10. At the moment the "date":1234567890 would be considered a year.
I think most of the time users are actually using the dates around our time, not so much in the future. So your suggestion that they should use epoch seconds/millis (or custom format) for such a dates makes sense to me.